### PR TITLE
DBZ-1065 Use separate connection for metadata

### PR DESCRIPTION
**Pull request by jpechane**
https://issues.jboss.org/projects/DBZ/issues/DBZ-1065

### CODE REVIEW

**Comment by gunnarmorling** [comment]
@jpechane, could you please shed some light on the reasoning? Ideally it should be part of the commit message. Thanks!

File: debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerStreamingChangeEventSource.java [hunk 1]
```diff
@@ -209,7 +212,8 @@ public void execute(ChangeEventSourceContext context) throws InterruptedExceptio
                             final Object[] dataNext = (operation == SqlServerChangeRecordEmitter.OP_UPDATE_BEFORE) ? tableWithSmallestLsn.getData() : null;
 
                             offsetContext.setChangePosition(tableWithSmallestLsn.getChangePosition(), eventCount);
-                            offsetContext.setSourceTime(connection.timestampOfLsn(tableWithSmallestLsn.getChangePosition().getCommitLsn()));
+                            //offsetContext.setSourceTime(Instant.ofEpochMilli(1000));
```
Comment (gunnarmorling) [inline debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerStreamingChangeEventSource.java]:
```suggestion
```

Comment (jpechane) [reply debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerStreamingChangeEventSource.java]: @gunnarmorling Changed, thanks

**Comment by jpechane** [comment]
@gunnarmorling The commit message has been updated. Generally the problem is that when adaptive buffering is in place then if there is a new uery issued while resultset is not consumed the JDBC driver will load the whole resultset into memory effectively ignoring the buffering.
This limitation is not present when the addtional query is issued on a separate connection.

**Comment by gunnarmorling** [comment]
Thanks, that makes sense!

@grzegorz8, any chance you give this a test ride and see whether it solves the issues you reported? Thanks a lot!

**Comment by gunnarmorling** [comment]
Added one commit for some clean-up and applied. Thanks, @jpechane!

**Comment by grzegorz8** [comment]
It's been working perfectly fine on production for at least a week now :-)

**Comment by gunnarmorling** [comment]
That's great to hear; thanks a lot for reporting back!

**Pull request merged**